sirbot/slack/store/user.py

```diff
@@ -56,16 +57,42 @@ class UserStore(SlackStore):
     def __init__(self, client, facades, refresh=3600):
         super().__init__(client, facades, refresh)
 
-    async def all(self):
+    async def all(self, fetch=False, deleted=False):
         db = self._facades.get('database')
-        data = await database.__dict__[db.type].user.get_all(db)
-
+        if fetch:
+            users = list()
+            fetched_data = await self._client.get_users()
+
+            for data in fetched_data:
+                user = User(
+                    id_=data['id'],
+                    raw=data['raw'],
+                    last_update=data['last_update'],
+                    dm_id=data['dm_id'],
+                    deleted=data['deleted'])
+
+                await database.__dict__[db.type].user.add(db, user)
+                if deleted:
+                    users.append(user)
+                if not deleted and not user.deleted:
+                    users.append(user)
+        else:
+            data = await database.__dict__[db.type].user.get_all(
+                db, deleted=deleted)
```

Comment:
Do you want to do the sql for this in [here ](https://github.com/pyslackers/sirbot-slack/blob/master/sirbot/slack/database/sqlite/user.py#L20) ? If not I can do it myself no problem :)